### PR TITLE
Add Student qr code for creating parent account

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-14197
affects: Student
release note: none

Testing:
1. enable flag `student_qr_code_pairing` 
2. Settings > Pair with observer
3. should show qr code 

4. disable same flag
5. make sure old pairing code flow shows up

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/j5878fv) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fj5878fv)
teacher: [Link](https://rebrand.ly/brzt2nz) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fbrzt2nz)
parent: [Link](https://rebrand.ly/cgh47hu) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fcgh47hu)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-14197](https://instructure.atlassian.net/browse/MBL-14197)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.5% | 93.5% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Profile/Settings/PairWithObserverViewController.swift [hunk 1]
```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
...
+        qrCodeImageView.image = UIImage(ciImage: qrImage)
+        qrCodeContainer.isHidden = false
+        codeContainer.isHidden = true
+        codeLabel.text = comps?.url?.absoluteString
+        deepLinkURL = comps?.url
+        let attrStr = NSAttributedString(
+            string: NSLocalizedString("Pairing Code: ", comment: ""),
+            attributes: [
+                NSAttributedString.Key.font: UIFont.scaledNamedFont(.regular20),
+                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest)
```
Comment (inst-danger) [inline Core/Core/Profile/Settings/PairWithObserverViewController.swift:129]:
```suggestion
                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest),
```

File: Core/Core/Profile/Settings/PairWithObserverViewController.swift [hunk 2]
```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
...
+            ]
+        )
+
+        let attrStr2 = NSAttributedString(
+            string: pairingCode ?? "",
+            attributes: [
+                NSAttributedString.Key.font: UIFont.scaledNamedFont(.semibold20),
+                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest)
```
Comment (inst-danger) [inline Core/Core/Profile/Settings/PairWithObserverViewController.swift:137]:
```suggestion
                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest),
```

File: Core/Core/Profile/Settings/PairWithObserverViewController.swift [hunk 3]
```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
...
+
+        var comps = URLComponents(string: "canvas-parent://create-account/create-account/\(accountID)/\(code)")
+        comps?.queryItems = [
+            URLQueryItem(name: "baseURL", value: host),
+        ]
+
+        let input = comps?.url?.absoluteString ?? ""
+        let data = input.data(using: String.Encoding.ascii)
+        guard let qrFilter = CIFilter(name: "CIQRCodeGenerator") else { return }
+        qrFilter.setValue(data, forKey: "inputMessage")
```
Comment (vanwagonet) [inline Core/Core/Profile/Settings/PairWithObserverViewController.swift:116]: I did not know it was so easy to generate a QR code. I love it.

File: Core/Core/Profile/Settings/PairWithObserverViewController.swift [hunk 4]
```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
...
+        let attrStr2 = NSAttributedString(
+            string: pairingCode ?? "",
+            attributes: [
+                NSAttributedString.Key.font: UIFont.scaledNamedFont(.semibold20),
+                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest),
+            ]
+        )
+        let mutableAttributedString = NSMutableAttributedString()
+        mutableAttributedString.append(attrStr)
+        mutableAttributedString.append(attrStr2)
```
Comment (vanwagonet) [inline Core/Core/Profile/Settings/PairWithObserverViewController.swift:142]: appending strings isn't great for i18n, but I don't know how else you'd do it for attributed strings, so this is fine for now.

File: Core/Core/Profile/Settings/PairWithObserverViewController.swift (hunk 3 above)
Comment (bluwave) [reply Core/Core/Profile/Settings/PairWithObserverViewController.swift:116]: I know, i was pretty surprised it was this easy myself.

**Pull request merged**